### PR TITLE
Bring the plugin back to date and add tests

### PR DESCRIPTION
**Pull request by purbon**
This PR basically takes care of bringing the plugin to be usable again with the last driver, latest releases of LS and adding small test to it.

### CODE REVIEW

File: lib/logstash/outputs/mongodb.rb [hunk 1]
```diff
@@ -62,11 +54,11 @@ def receive(event)
       if @generateId
         document['_id'] = BSON::ObjectId.new(nil, event["@timestamp"])
       end
-      @db.collection(event.sprintf(@collection)).insert(document)
+      @db[event.sprintf(@collection)].insert_one(document)
     rescue => e
       @logger.warn("Failed to send event to MongoDB", :event => event, :exception => e,
                    :backtrace => e.backtrace)
-      if e.error_code == 11000
+      if e.message =~ /^E11000/
```
Comment (guyboertje) [inline lib/logstash/outputs/mongodb.rb:60]: +1 on removing the specific method call to `#error_code`

Comment (purbon) [inline lib/logstash/outputs/mongodb.rb:60]: ++

File: lib/logstash/outputs/mongodb.rb [hunk 2]
```diff
@@ -34,17 +34,9 @@ class LogStash::Outputs::Mongodb < LogStash::Outputs::Base
   public
   def register
     require "mongo"
```
Comment (guyboertje) [inline lib/logstash/outputs/mongodb.rb]: how do you feel about moving the require up to the top? ala https://github.com/elastic/logstash/pull/3404#issuecomment-113170046

Comment (purbon) [inline lib/logstash/outputs/mongodb.rb]: that makes sense

**Comment by jsvd** [comment]
Unit tests run, didn't do an integration test, but code LGTM

**Pull request closed without merging**